### PR TITLE
add traceability agent audit logs

### PR DESCRIPTION
**Pull request by jcollins-axway**
## Describe the changes

Add new page detailing the audit logs in the traceability agents.

## Deploy preview link

http://pr-438.amplify-central.opendocs-builder.pcloud.axway.int/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs/

## Checklist for contributors

Before submitting this PR:

* Verify that all status checks have passed
* For more information on the Markdown linter rules, see [DavidAnson/markdownlint](https://github.com/DavidAnson/markdownlint)
* For first-time contributors, ensure that you have signed the [Axway CLA](https://cla.axway.com/)

### CODE REVIEW

**Comment by docsbuilder** [comment]
### <span aria-hidden="true">✅</span> Deploy preview is ready!

 |  Name | Link |
 |---------------------------------|------------------------|
 |<span aria-hidden="true">🔨</span> Latest commit | dd3698d |
 |<span aria-hidden="true">😎</span> Deploy preview | http://PR-438.amplify-central.opendocs-builder.pcloud.axway.int |

**Comment by jcollins-axway** [comment]
@sbolosan and @vivekschauhan could you check for content?

**Review by vivekschauhan — APPROVED** [review]

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ The audit log in these cases would look like the following.
 {"id":"2fc1f178-abb3-4b3f-8edc-41edc6932c70","level":"info","message":"published","time":"2024-04-19T22:31:08Z"}
 ```
 
-Notice that each metric event is linked to the same api, however other fields have adjusted for new transactions. A field to note is the startTimestamp, in both cases the observation period for the event was at the same time. This is another clue that the metric counts from the first event were sent with the second event. 
+Notice that each metric event is linked to the same api, however other fields have adjusted for new transactions. A field to note is the startTimestamp, in both cases the observation period for the event was at the same time. This is another clue that the metric counts from the first event were sent with the second event.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:106]: A field to note is the 'startTimestamp'. In both cases, the observation period...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+title: Understand the Traceability Agent audit logs
+linkTitle: Traceability Agent audit logs
+draft: false
+weight: 10
+---
+The Traceability Agent has the ability to keep audit logs for tracking information about what the agent has processed.  The following requires agents built with Agent SDK v1.1.80 or later.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:7]: The following requires that agents be build with Agent SDK v1.1.80 or later

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Learn how to configure, access, and read the Traceability agent audit logs.
+
+* Configure - learn about the configuration options for the audit logs
+* Access - learn how to start the agent to mount the logs directory
+* Understand - learn how to read the audit logs the agent produces
+
+## Configure
+
+In all traceability agents the metric audit log is turned on by default it is used to track all metric events that the agent has generated and the status of that event as known to the agent. The following tables lists the environment variables available to configure the metric audit logs.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:19]: In all traceability agents, the metric audit log is turned on by default.  It ...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+| Environment Variable            | Description                                                                         |
+| ------------------------------- | ----------------------------------------------------------------------------------- |
+| LOG_METRICFILE_ENABLED          | Enable or disable the metric audit log (default: `true`)                            |
+| LOG_METRICFILE_NAME             | The name of the metric log file that will be created (default: `metric.log`)        |
+| LOG_METRICFILE_KEEPFILES        | The maximum number of metric audit log files to keep (default: `7`)                 |
+| LOG_METRICFILE_ROTATEEVERYBYTES | The maximum size, in megabytes a metric audit log can grow to (default: `10485760`) |
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:26]: The maximum size (in megabytes) a metric audit...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+| LOG_METRICFILE_CLEANBACKUPS     | The max age of a metric audit log backup file, in days (default: `0` unlimited)     |
+
+With the default settings the traceability agent will keep, at most, `7` files each a total of `10485760` megabytes.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:29]: With the default settings, the traceability...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+### Additional Axway API Manager Traceability agent audit log
+
+The Traceability agent that connects to Axway API Manager includes an additional transaction audit log to track the correlation IDs of each transaction the agent has seen and recorded. Here are the environment variables available to configure the transaction audit logs.
+
+| Environment Variable             | Description                                                                              |
+| -------------------------------- | ---------------------------------------------------------------------------------------- |
+| LOG_TRANSACTION_ENABLED          | Enable or disable the transaction audit log (default: `true`)                            |
+| LOG_TRANSACTION_NAME             | The name of the transaction log file that will be created (default: `transaction.log`)   |
+| LOG_TRANSACTION_KEEPFILES        | The maximum number of transaction audit log files to keep (default: `7`)                 |
+| LOG_TRANSACTION_ROTATEEVERYBYTES | The maximum size, in megabytes a transaction audit log can grow to (default: `10485760`) |
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:40]: The maximum size (in megabytes) a transaction...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+All audit logs will be created within an `audit` directory directly underneath the `LOG_FILE_PATH` path.
+
+### Binary
+
+The audit logs will be within an `audit` directory found in the logs directory.
+
+### Docker agents
+
+For all docker agents a new volume has been added, which when mounted will save the log directory outside of the container. All that is needed is to supply a volume argument to the start command `-v /path/outside/container:/logs`.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:53]: For all docker agents, a new volume...
which, when mounted, will save...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+### Helm agents
+
+For agents that a helm chart is supplied, Axway API Manager Traceability agent, a new Persistent Volume Claim is created with the helm chart which will be used for the logs directory.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:57]: For agents that have a helm chart supplied, ....
... is created with the helm chart, which will be used...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+The metric audit log is used to track the metric events the agent has created based off traffic through the gateway. This log file will generate 1..n lines for each metric event to help track the progress of that metric.
+
+The first line for each metric event is the `generated` line. This line gives all the pertinent information about a particular metric event, including the `id`. The `id` field is important to note as the follow up log lines will not include details about the metric event, only its id and a message about its state.
+
+```bash
+{"apiID":"remoteApiId_store","avgResponse":8,"count":1,"endTimestamp":1713556832244,"id":"47293432-7f9a-4f12-ba67-2338be048cb0","level":"info","maxResponse":8,"message":"generated","minResponse":8,"startTimestamp":1713556375134,"status":"403","time":"2024-04-19T20:00:32Z"}
+```
+
+After an event is `generated` the following types of events may be seen, giving information about the state of that metric event.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:73]: After an event is `generated`, ...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+For instance in a good path situation the audit log will have the following.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:75]: For instance, in a good path...

File: content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+{"id":"ea2be3b3-5931-47b2-b5eb-851036b9b9f0","level":"info","message":"publishing","time":"2024-04-12T07:47:35-07:00"}
+{"id":"ea2be3b3-5931-47b2-b5eb-851036b9b9f0","level":"info","message":"event cancelled, counts added at next publish","time":"2024-04-12T07:47:35-07:00"}
+{"apiID":"remoteApiId_44dc9d85-c1d8-465f-a7ff-88cb9a631f6e","avgResponse":1318,"count":100,"endTimestamp":1712933652207,"id":"ea2be3b3-5931-47b2-b5eb-851036b9b9f0","level":"info","maxResponse":1318,"message":"generated","minResponse":1318,"startTimestamp":1712932910655,"status":"200","time":"2024-04-12T07:54:12-07:00"}
+{"id":"ea2be3b3-5931-47b2-b5eb-851036b9b9f0","level":"info","message":"publishing","time":"2024-04-12T07:54:24-07:00"}
+{"id":"ea2be3b3-5931-47b2-b5eb-851036b9b9f0","level":"info","message":"published","time":"2024-04-12T07:54:25-07:00"}
+```
+
+As can be seen above all lines contain the same `id`, new `generated` events take precedence over any previous events. These are the events that will be reflected in Amplify.
```
Comment (sbolosan) [inline content/en/docs/connect_manage_environ/connected_agent_common_reference/traceability_audit_logs.md:93]: As can be seen from above, all lines contain...

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Comment by lbadenhop** [comment]
@jcollins-axway, please review my edits before I merge.

**Comment by jcollins-axway** [comment]
> @jcollins-axway, please review my edits before I merge.

looks great, thank you!

**Pull request merged**